Source/ARTHttp.m

```diff
@@ -175,18 +179,19 @@ - (void)executeRequest:(NSMutableURLRequest *)request completion:(void (^)(NSHTT
     [self.logger debug:@"%@ %@", request.HTTPMethod, request.URL.absoluteString];
     [self.logger verbose:@"Headers %@", request.allHTTPHeaderFields];
 
+    __weak typeof(self) weakSelf = self;
```

Comment:
> When an ARTHttp object gets deallocated but the request is still active.

Yes, but why? I mean, if we use a normal strong reference in the async callback as we did before, the ARTHttp object won't be deallocated until the callback is called, ie. while the request is active. Why would you want to prevent that?